php/symfony/src/Controller/HomeController.php

```diff
@@ -2,15 +2,61 @@
 
 namespace App\Controller;
 
+use Doctrine\DBAL\Exception;
+use Doctrine\ORM\EntityManagerInterface;
+use League\Flysystem\FilesystemException;
+use League\Flysystem\FilesystemOperator;
+use Psr\Cache\CacheItemPoolInterface;
+use Psr\Cache\InvalidArgumentException;
+use Psr\Log\LoggerInterface;
 use Symfony\Component\Routing\Annotation\Route;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\Response;
 
 class HomeController extends AbstractController
 {
     #[Route('/', name: 'home')]
-    public function home(): Response
+    public function home(EntityManagerInterface $orm, CacheItemPoolInterface $app, FilesystemOperator $persistentStorage, LoggerInterface $logger): Response
     {
-        return $this->render('deploio.html.twig');
+
+        try {
+            $status['database'] = is_string($orm->getConnection()->getDatabase());
+        } catch (Exception $e) {
+            $logger->warning('Error while trying to use database: "{message}"', [
+                'message' => $e->getMessage(),
+                'exception' => $e,
+            ]);
+            $status['database'] = false;
+        }
+
+        try {
+            $item = $app->getItem('foo');
+            $item->set('check');
+            $app->save($item);
+            // if the cache is not working, there is no exception to not interrupt the program flow, but nothing is stored
+            $status['kvs'] = $app->hasItem('foo');
+        } catch (InvalidArgumentException $e) {
+            $logger->warning('Error while trying to use key value store: "{message}"', [
+                'message' => $e->getMessage(),
+                'exception' => $e,
+            ]);
+            $status['kvs'] = false;
+        }
+
+        try {
+            $persistentStorage->write('/test-file', 'Symfony Test');
```

Comment:
The variable naming might be a bit confusing as deplo.io currently doesn't provide real persistent storage. After every release, the filesystem will have a clear state. So one can use it for temporary storage, but for persistent storage the object store might be the better choice.

Reply:
the tutorial is setting this up with s3 - my understanding was that the s3 service would provide persistent storage?

Reply:
Ah sorry, I was under the impression that this refers to file system storage instead of s3. All good then.